patches/react-native-onyx+1.0.15.patch

```diff
@@ -0,0 +1,175 @@
+diff --git a/node_modules/react-native-onyx/lib/fastMerge.js b/node_modules/react-native-onyx/lib/fastMerge.js
+new file mode 100644
+index 0000000..f3adbc8
+--- /dev/null
++++ b/node_modules/react-native-onyx/lib/fastMerge.js
+@@ -0,0 +1,97 @@
++/**
++ * Is the object Mergeable
++ *
++ * @param val
++ * @returns {*|boolean}
++ */
++ function isMergeableObject(val) {
++  var nonNullObject = val && typeof val === 'object';
++  return nonNullObject &&
++      Object.prototype.toString.call(val) !== '[object RegExp]' &&
++      Object.prototype.toString.call(val) !== '[object Date]';
++}
++/**
++* Empty the Target
++*
++* @param val
++* @returns {*}
++*/
++function emptyTarget(val) {
++  return Array.isArray(val) ? [] : {}
++}
++/**
++* Clone if Necessary
++*
++* @param value
++* @param optionsArgument
++* @returns {*}
++*/
++function cloneIfNecessary(value, optionsArgument) {
++  var clone = optionsArgument && optionsArgument.clone === true;
++  return (clone && isMergeableObject(value)) ?     merge(emptyTarget(value), value, optionsArgument) : value;
++}
++/**
++* Default Array Merge
++*
++* @param target
++* @param source
++* @param optionsArgument
++* @returns {*}
++*/
++let valArr;
++function defaultArrayMerge(target, source, optionsArgument) {
++  var destination = target.slice();
++  for (let i = 0; i < source.length; ++i) {
++    valArr = source[i];
++    if (i >= destination.length) {
++      destination.push(cloneIfNecessary(valArr, optionsArgument));
++    } else if (typeof destination[i] === 'undefined') {
++      destination[i] = cloneIfNecessary(valArr, optionsArgument);
++    } else if (isMergeableObject(valArr)) {
++      destination[i] = merge(target[i], valArr, optionsArgument);
++    }
++  }
++  return destination;
++}
++/**
++* Merge Object
++*
++* @param target
++* @param source
++* @param optionsArgument
++* @returns {{}}
++*/
++function mergeObject(target, source, optionsArgument) {
++  var destination = {};
++  if (isMergeableObject(target)) {
++      Object.keys(target).forEach(function (key) {
++          destination[key] = cloneIfNecessary(target[key], optionsArgument)
++      })
++  }
++  Object.keys(source).forEach(function (key) {
++      if (!isMergeableObject(source[key]) || !target[key]) {
++          destination[key] = cloneIfNecessary(source[key], optionsArgument)
++      } else {
++          destination[key] = merge(target[key], source[key], optionsArgument)
++      }
++  });
++  return destination
++}
++/**
++* Merge Object and Arrays
++*
++* @param target
++* @param source
++* @param optionsArgument
++* @returns {*}
++*/
++export function merge(target, source, optionsArgument) {
++  var array = Array.isArray(source);
++  var options = optionsArgument || {arrayMerge: defaultArrayMerge};
++  var arrayMerge = options.arrayMerge || defaultArrayMerge;
++  if (array) {
++      return Array.isArray(target) ? arrayMerge(target, source, optionsArgument) : cloneIfNecessary(source, optionsArgument);
```

Comment:
Can you explain this logic a little bit? I'm not sure that I understand the intention of it.

Reply:
Basically if there is no array then just clone the one from source. If there is then merge every index separately. 
Not sure if merging indices is necessary here but _.merge merges them so that's why it's here. I changed only arrayMerge as it was O(n^2) the rest is copied from the website I linked above.

Reply:
Ok this makes a lot of sense to me just really odd lodash would not use this approach as well.

@Szymon20000 is there any case this method handles differently/ returns different result than lodash merge?